source/Calamari.AzureAppService/Behaviors/TargetDiscoveryBehaviour.cs

```diff
@@ -1,25 +1,35 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Collections.ObjectModel;
 using System.Linq;
 using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading;
 using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.ResourceManager;
+using Azure.ResourceManager.AppService;
+using Azure.ResourceManager.Models;
+using Azure.ResourceManager.Resources;
 using Calamari.AzureAppService.Azure;
 using Calamari.Common.Commands;
 using Calamari.Common.Features.Discovery;
 using Calamari.Common.Plumbing.Logging;
 using Calamari.Common.Plumbing.Pipeline;
 using Calamari.Common.Plumbing.ServiceMessages;
 using Calamari.Common.Plumbing.Variables;
-using Microsoft.Azure.Management.AppService.Fluent;
-using Microsoft.Azure.Management.AppService.Fluent.Models;
-using Microsoft.Azure.Management.Fluent;
-using Polly;
+using JsonException = System.Text.Json.JsonException;
+using JsonSerializer = System.Text.Json.JsonSerializer;
 
 #nullable enable
 namespace Calamari.AzureAppService.Behaviors
 {
     public class TargetDiscoveryBehaviour : IDeployBehaviour
     {
+        private const string WebAppSlotsType = "sites/slots";
+        private const string WebAppType = "sites";
+        private const int PageSize = 500;
```

Comment:
If there are more than 500 resources are we handling that anywhere? Or is it handled by the client?

Reply:
I tested this, the PageSize we pass in doesn't stop us from getting all the resources if we simply iterate the async collection, it will actually get us all the resources but it will request them in batches of PageSize when it does it. It's really nice and clean for us 👍

Reply:
Ah nice one 👍 how did we settle on a page size of 500?

Reply:
To be honest I just kinda napkin mathed it - knowing we had ~250 results coming back from a single request and that took ~1-1.5 seconds I thought double that would be the max I would want to query for - maybe that's too long? Paging too low does seem to slow down the process overall. I did some tests where I set it to 10, 100, 200 and 10 and 100 were slower, 200 was pretty good, a single request was the fastest. I guess that makes sense. But for someone with 1000 web apps/slots I don't know what would be best.

Reply:
Hmmm yeah I think that seems reasonable given what you've described there, we want to balance the number of requests with the size of each request and try to find something nice in the middle. Do the azure docs for the sdk have any guidance?

Reply:
We have also started to use the sdk in other places too right? Do we use paged queries anywhere there that could give us a guide on what a good number could be?

Reply:
> Do the azure docs for the sdk have any guidance?

I've looked around a bit and I couldn't find any recommendations on page size. I guess it depends on the query you're making. For instance the performance is a lot worse if you're filtering by tags so you probably want a lower page size. When filtering by resource type, it's quite quick. 

> We have also started to use the sdk in other places too right? Do we use paged queries anywhere there that could give us a guide on what a good number could be?

It's only used in one other place and unfortunately for us, no paging is used.

Reply:
> I've looked around a bit and I couldn't find any recommendations on page size. I guess it depends on the query you're making. For instance the performance is a lot worse if you're filtering by tags so you probably want a lower page size. When filtering by resource type, it's quite quick.

Yeah makes sense, I think based on your testing so far 500 seems reasonable, if it starts causing problems we can revisit.